files/bootstrap.sh

```diff
@@ -392,24 +399,40 @@ Environment='KUBELET_EXTRA_ARGS=$KUBELET_EXTRA_ARGS'
 EOF
 fi
 
-# Replace with custom docker config contents.
-if [[ -n "$DOCKER_CONFIG_JSON" ]]; then
+if [[ "$CONTAINER_RUNTIME" = "containerd" ]]; then
+    sudo mkdir -p /etc/containerd
+    sudo mkdir -p /etc/cni/net.d
+    sudo mv /etc/eks/containerd/containerd-config.toml /etc/containerd/config.toml
+    sudo mv /etc/eks/containerd/kubelet-containerd.service /etc/systemd/system/kubelet.service
+    sudo chown root:root /etc/systemd/system/kubelet.service
+    systemctl daemon-reload
+    systemctl enable containerd
+    systemctl start containerd
+elif [[ "$CONTAINER_RUNTIME" = "dockerd" ]]; then
     mkdir -p /etc/docker
-
-    echo "$DOCKER_CONFIG_JSON" > /etc/docker/daemon.json
-    systemctl restart docker
+    bash -c "/sbin/iptables-save > /etc/sysconfig/iptables"
+    mv /etc/eks/iptables-restore.service /etc/systemd/system/iptables-restore.service
+    sudo chown root:root /etc/systemd/system/iptables-restore.service
+    systemctl daemon-reload
+    systemctl enable iptables-restore
+
+    if [[ -n "$DOCKER_CONFIG_JSON" ]]; then
+        echo "$DOCKER_CONFIG_JSON" > /etc/docker/daemon.json
+    fi
+    if [[ "$ENABLE_DOCKER_BRIDGE" = "true" ]]; then
+          # Enabling the docker bridge network. We have to disable live-restore as it
+          # prevents docker from recreating the default bridge network on restart
+          echo "$(jq '.bridge="docker0" | ."live-restore"=false' /etc/docker/daemon.json)" > /etc/docker/daemon.json
+    fi
+    systemctl daemon-reload
+    systemctl enable docker
+    systemctl start docker
```

Comment:
Hi
In the previous version of this file the Docker daemon was always restarted after its config file was modified.  But after this commit nothing happens, the changes don't have an effect, probably because now the script runs `systemctl start docker` instead of `systemctl restart docker`

Reply:
This change is impacting our automated testing too. Was wondering what the rationale between switching from `restart` to `start` was.